platform/admin/core/admin-core-insecuredefaults/src/main/java/org/codice/ddf/admin/insecure/defaults/service/DefaultUsersDeletionScheduler.java

```diff
@@ -0,0 +1,234 @@
+        return null;
+      }
+
+      if (Instant.now()
+          .isAfter(
+              firstInstall
+                  .plus(Duration.ofDays(3))
+                  .minus(Duration.ofMinutes(30)))) { // Timestamp expired
+        removeDefaultUsers();
+        return null;
+      }
+
+      return cronCalculator(firstInstall);
+    }
+  }
+
+  private boolean createTempFile(Instant instant) {
+    try {
+      if (getTempTimestampFilePath().toFile().createNewFile()) {
+        return writeTimestamp(instant);
+      }
+    } catch (IOException e) {
+      LOGGER.debug("Unable to access the temporary file", e);
+    }
+    return false;
+  }
+
+  private boolean writeTimestamp(Instant instant) throws IOException {
+    try (ObjectOutputStream objectOutputStream =
+        new ObjectOutputStream(new FileOutputStream(getTempTimestampFilePath().toFile()))) {
+      objectOutputStream.writeObject(instant);
+      return true;
+    }
+  }
+
+  public static boolean removeDefaultUsers() {
+    try {
+      Bundle bundle = FrameworkUtil.getBundle(DefaultUsersDeletionScheduler.class);
+      if (bundle != null) {
+        BundleContext bundleContext = bundle.getBundleContext();
+        Collection<ServiceReference<BackingEngineFactory>> implementers =
+            bundleContext.getServiceReferences(BackingEngineFactory.class, null);
+        for (ServiceReference<BackingEngineFactory> impl : implementers) {
+
+          BackingEngineFactory backingEngineFactory = bundleContext.getService(impl);
+          BackingEngine backingEngine =
+              backingEngineFactory.build(
+                  ImmutableMap.of("users", getUsersPropertiesFilePath().toString()));
+
+          if (!backingEngine.listUsers().isEmpty()) {
+            backingEngine.listUsers().forEach(user -> backingEngine.deleteUser(user.getName()));
+          }
+        }
+      }
+
+      LOGGER.debug("Default users have been deleted successfully.");
+
+      Files.deleteIfExists(getTempTimestampFilePath());
+      return true;
+    } catch (Exception e) {
+      LOGGER.debug("Unable to remove default users.", e);
+      return false;
+    }
+  }
+
+  private String cronCalculator(Instant firstInstall) {
+    Instant threeDayTimestamp = firstInstall.plus(Duration.ofDays(3).minus(Duration.ofMinutes(30)));
+    LocalDateTime localDateTime =
+        LocalDateTime.ofInstant(threeDayTimestamp, ZoneId.systemDefault());
+
+    return String.format(
+        "%d+%d+%d+%d+%d+?+%d",
+        localDateTime.getSecond(),
+        localDateTime.getMinute(),
+        localDateTime.getHour(),
+        localDateTime.getDayOfMonth(),
+        localDateTime.getMonthValue(),
+        localDateTime.getYear());
+  }
+
+  public Instant installationDate() {
+    if (!getTempTimestampFilePath().toFile().exists()) {
+      return null;
+    }
+    try (ObjectInputStream objectInputStream =
+        new ObjectInputStream(new FileInputStream(getTempTimestampFilePath().toFile()))) {
```

Comment:
❗️  Doesn't using java object serialization open avenues for malicious injection ?

Is there a reason we did not just serialize as ISO8601 or some other standard encoding for a date?

Reply:
@rzwiefel, could you explain how malicious injection could occur here? I'm just not seeing it.

Reply:
https://www.owasp.org/index.php/Deserialization_of_untrusted_data

Reply:
We can address this in another PR if needed.